lib/queuePopulator/IngestionProducer.js

```diff
@@ -69,6 +70,25 @@ class IngestionProducer {
         ], err => done(err, this.resLog));
     }
 
+    getRaftLog(raftId, begin, end, targetLeader, done) {
+        const req = this.ringReader.getRaftLog({
+            LogId: raftId,
+            Begin: begin,
+            End: end,
+            TargetLeader: targetLeader,
+        });
+        attachReqUids(req, this.requestLogger);
+        const readStream = req.createReadStream();
+        const jsonResponse = readStream.pipe(jsonStream.parse('log.*'));
```

Comment:
actually, even delete object entries don't have a "put" field specified.. hmm

Reply:
You're right, I should have the transform here to ensure that `put` is applied if there is no specified `entry.type`.

I don't think the the `logReader` side needs the entry type specification

Reply:
Isn't this because of using `console.log` to log a javascript response object? `console.log` prunes inner objects reaching some depth so will not log everything, but it can be prior `JSON.stringify` for this purpose.

Reply:
The `type` field is at the same level (or should be) as `key` and `value` if it exists, as per the metadata documentation https://github.com/scality/MetaData/blob/development/8.0/documentation/PROTOCOL.ADMIN.md#get-raft-session-logs-2
so it shouldn't be the inner object issue?

Reply:
even with `JSON.stringify`, i'm seeing the following from a `send` response
`{"db":"something","entries":[{"key":"bay1"}],"method":8}`
and the following from stream:
`{"db":"something","entries":[{"key":"bay1","type":"del"}],"method":8}`

I'm most likely doing something wrong in my script x.x If I figure it out, I'll mention it here